src/lib.rs

```diff
@@ -72,10 +74,27 @@ pub use crate::{
 /// # }
 /// ```
 pub struct LockFile {
+    pub truncate_on_close: bool,
     locked: bool,
     desc: sys::FileDesc,
 }
 
+// Private functions
+impl LockFile {
+    fn new() -> Self {
+        #[allow(unused_unsafe)]
+        // not safe to get state of default truncation setting on no_std
+        Self {
+            desc: sys::uninitialized_fd(),
+            locked: false,
+            truncate_on_close: unsafe {
+                constants::default_lockfile_truncate_state()
+            },
```

Comment:
As I said, don't use global state. I'd rather just have `true` as the default (note that `false` as the default would break the API).